controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -108,7 +108,7 @@ func (w *Workload) updateManagedEtcdConditions(ctx context.Context, controlPlane
 			conditions.MarkUnknown(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberInspectionFailedReason, "Failed to connect to the etcd pod on the %s node: %s", node.Name, err)
 			continue
 		}
-		defer etcdClient.Close()
+		defer etcdClient.Close() //nolint:gocritic // etcdClient is closed correctly, no resource leak.
```

Comment:
Is this the same issue as https://github.com/kubernetes-sigs/cluster-api/pull/5789 ? Same for the other instances of .Close here.

Reply:
It could be better to move this for loop in a function and call that instead of deferring in a for loop?

Reply:
Done